sentry_sdk/integrations/ariadne.py

```diff
@@ -0,0 +1,175 @@
+
+        if hub.client:
+            with capture_internal_exceptions():
+                for error in errors:
+                    event, hint = event_from_exception(
+                        error,
+                        client_options=hub.client.options,
+                        mechanism={
+                            "type": hub.get_integration(AriadneIntegration).identifier,
+                            "handled": False,
+                        },
+                    )
+                    hub.capture_event(event, hint=hint)
+
+        return result
+
+    def _sentry_patched_handle_query_result(result, *args, **kwargs):
+        # type: (Any, Any, Any) -> GraphQLResult
+        hub = Hub.current
+        integration = hub.get_integration(AriadneIntegration)
+        if integration is None:
+            return old_handle_query_result(result, *args, **kwargs)
+
+        response = old_handle_query_result(result, *args, **kwargs)
+
+        with hub.configure_scope() as scope:
+            event_processor = _make_response_event_processor(response[1])
+            scope.add_event_processor(event_processor)
+
+        if hub.client:
+            with capture_internal_exceptions():
+                for error in result.errors or []:
+                    event, hint = event_from_exception(
+                        error,
+                        client_options=hub.client.options,
+                        mechanism={
+                            "type": hub.get_integration(AriadneIntegration).identifier,
+                            "handled": False,
+                        },
+                    )
+                    hub.capture_event(event, hint=hint)
+
+        return response
+
+    ariadne_graphql.parse_query = _sentry_patched_parse_query  # type: ignore
+    ariadne_graphql.handle_graphql_errors = _sentry_patched_handle_graphql_errors  # type: ignore
+    ariadne_graphql.handle_query_result = _sentry_patched_handle_query_result  # type: ignore
+
+
+def _make_request_event_processor(data):
+    # type: (GraphQLSchema) -> EventProcessor
+    """Add request data and api_target to events."""
+
+    def inner(event, hint):
+        # type: (Dict[str, Any], Dict[str, Any]) -> Dict[str, Any]
+        if not isinstance(data, dict):
+            return event
+
+        with capture_internal_exceptions():
+            try:
+                content_length = int(
+                    (data.get("headers") or {}).get("Content-Length", 0)
+                )
+            except (TypeError, ValueError):
+                return event
+
+            if _should_send_default_pii() and request_body_within_bounds(
+                Hub.current.client, content_length
+            ):
+                request_info = event.setdefault("request", {})
+                request_info["api_target"] = "graphql"
+                request_info["data"] = data
+
+            elif event.get("request", {}).get("data"):
+                del event["request"]["data"]
```

Comment:
is this set somewhere outside of the integrations? and if yes, are we overwriting it in line 150?

Reply:
The event processors from our web framework integrations add this (tested at least with Flask and FastAPI). So line 150 would be overwriting stuff if something was already there. I'll make it conditional, i.e., only set `data` if not set (so that you also get the request data if you don't have a suitable web framework integration enabled).